idaes/core/initialization/block_triangularization.py

```diff
@@ -168,6 +167,12 @@ def _solve_block_data(self, block_data, solver):
         """
         Call solve_strongly_connected_components on a given BlockData.
         """
+        if number_activated_constraints(block_data) == 0:
```

Comment:
Why is this here? Should there be some kind of warning or error thrown?

Reply:
This is here because Pyomo throws an error when `solve` is called on a block with no constraints. We need to support initialization for blocks with no constraints because block triangularization is the default initializer for state blocks. Certain state blocks end up having no constraints (typically when `defined_state == True`). It's easier to put this check in the block triangularization rather than building out infrastructure for every property package to handle this circumstance.